File: suite/Evenodd/Evenodd.rs
fn odd_abs(n: u64) -> bool {
    if n == 0 {
        false
    } else {
        even_abs(n - 1)
    }
}

fn even_abs(n: u64) -> bool {
    if n == 0 {
        true
    } else {
        odd_abs(n - 1)
    }
}

fn odd(n: i64) -> bool {
    odd_abs(n.abs() as u64)
}

fn even(n: i64) -> bool {
    even_abs(n.abs() as u64)
}

fn main_loop(iters: u64, n: i64) {
    let res = even(n) && !odd(n);
    if iters == 1 {
        if res {
            println!("1")
        } else {
            println!("0")
        }
    } else {
        main_loop(iters - 1, n)
    }
}

fn main() {
    let mut args = std::env::args();
    args.next();
    let iters = args
        .next()
        .expect("Missing Argument iterations")
        .parse::<u64>()
        .expect("Iterations must be a number");
    let n = args
        .next()
        .expect("Missing Argument n")
        .parse::<i64>()
        .expect("n must be a number");
    main_loop(iters, n)
}

File: suite/Evenodd/Evenodd.ml
let abs_int i =
  if i < 0 then -i else i

let rec odd_abs n =
  if n = 0 then false else even_abs (n - 1)
and even_abs n =
  if n = 0 then true else odd_abs (n - 1)

let even n = even_abs (abs_int n)
let odd n = odd_abs (abs_int n)

let rec main_loop iters n =
  let res = even n && (not (odd n)) in
  if iters = 1 then
    if res then print_endline "1" else print_endline "0"
  else main_loop (iters - 1) n

let main =
  let iters = int_of_string Sys.argv.(1) in
  let n = int_of_string Sys.argv.(2) in
  main_loop iters n
